File: game.cpp
#include <iostream>
#include <string>
#include <vector>
#include <random>
#include <algorithm>

struct Item {
    std::string name;
    double price;
};

int main() {
    // Seed for random number generation
    std::random_device rd;
    std::mt19937 gen(rd());

    // List of items with their prices
    std::vector<Item> items = {
        {"Toaster", 19.99},
        {"Blender", 49.99},
        {"Microwave Oven", 89.99},
        {"Vacuum Cleaner", 129.99},
        {"Coffee Maker", 39.99},
        {"Smartphone", 699.99},
        {"Laptop", 999.99},
        {"Headphones", 199.99},
        {"Smart Watch", 249.99},
        {"Tablet", 329.99}
    };

    // Select a random item
    std::uniform_int_distribution<> itemDist(0, items.size() - 1);
    Item selectedItem = items[itemDist(gen)];

    // Generate price options
    std::vector<double> priceOptions;
    priceOptions.push_back(selectedItem.price); // Correct price

    // Generate three incorrect prices
    std::uniform_real_distribution<> priceVariance(-0.5, 0.5);
    while (priceOptions.size() < 4) {
        double variance = priceVariance(gen);
        double incorrectPrice = selectedItem.price + (selectedItem.price * variance);

        // Round to two decimal places
        incorrectPrice = std::round(incorrectPrice * 100.0) / 100.0;

        // Ensure the incorrect price is not equal to the correct price and not already in the list
        if (incorrectPrice != selectedItem.price &&
            std::find(priceOptions.begin(), priceOptions.end(), incorrectPrice) == priceOptions.end()) {
            priceOptions.push_back(incorrectPrice);
        }
    }

    // Shuffle the price options
    std::shuffle(priceOptions.begin(), priceOptions.end(), gen);

    // Display the game prompt
    std::cout << "Guess the correct price of the following item:\n";
    std::cout << "Item: " << selectedItem.name << "\n\n";
    std::cout << "Price Options:\n";
    for (size_t i = 0; i < priceOptions.size(); ++i) {
        std::cout << i + 1 << ") $" << priceOptions[i] << "\n";
    }

    // Get player's guess
    int choice;
    std::cout << "\nEnter the number corresponding to your guess: ";
    std::cin >> choice;

    // Validate input
    if (choice < 1 || choice > static_cast<int>(priceOptions.size())) {
        std::cout << "Invalid choice. Please run the program again.\n";
        return 1;
    }

    // Check if the guess is correct
    if (priceOptions[choice - 1] == selectedItem.price) {
        std::cout << "Congratulations! You guessed the correct price.\n";
    } else {
        std::cout << "Sorry, that's incorrect. The correct price was $" << selectedItem.price << ".\n";
    }

    return 0;
}

File: game.py
import random

class Item:
    def __init__(self, name, price):
        self.name = name
        self.price = price

def get_items():
    return [
        Item("Toaster", 19.99),
        Item("Blender", 49.99),
        Item("Microwave Oven", 89.99),
        Item("Vacuum Cleaner", 129.99),
        Item("Coffee Maker", 39.99),
        Item("Smartphone", 699.99),
        Item("Laptop", 999.99),
        Item("Headphones", 199.99),
        Item("Smart Watch", 249.99),
        Item("Tablet", 329.99)
    ]

def contestants_bids(num_contestants, min_bid, max_bid):
    bids = []
    for i in range(num_contestants):
        while True:
            try:
                bid = float(input(f"Contestant {i+1}, enter your bid between ${min_bid:.2f} and ${max_bid:.2f}: $"))
                if min_bid <= bid <= max_bid:
                    bids.append(bid)
                    break
                print("Bid out of range.")
            except ValueError:
                print("Invalid input. Enter a number.")
    return bids

def main():
    print("Welcome to The Price is Right!\n")
    items = get_items()
    selected_item = random.choice(items)
    print(f"Today's item: {selected_item.name}\n")

    num_contestants = 4
    min_bid = 1.00
    max_bid = selected_item.price * 2

    print(f"Actual retail price is hidden. Four contestants will bid as close as possible without going over!\n")
    bids = contestants_bids(num_contestants, min_bid, max_bid)

    print("\nBids:")
    for i, bid in enumerate(bids):
        print(f"Contestant {i+1}: ${bid:.2f}")

    # Find the highest bid not over the price
    winning_index = -1
    winning_bid = -1
    for i, bid in enumerate(bids):
        if bid <= selected_item.price and bid > winning_bid:
            winning_bid = bid
            winning_index = i

    print(f"\nActual retail price: ${selected_item.price:.2f}")
    if winning_index == -1:
        print("All contestants overbid! No winner this round.")
    else:
        print(f"Contestant {winning_index+1} wins with a bid of ${winning_bid:.2f}!")

if __name__ == "__main__":
    main()
